File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head: return head
        
        tmpPrev = None
        tmpNext = None
        curr = head
        
        while curr:
            tmpNext = curr.next
            curr.next = tmpPrev
            tmpPrev = curr
            curr = tmpNext
            
            
        curr = tmpPrev
        
        return curr

File: 0206-reverse-linked-list/0206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let tmpPrev = null;
    let tmpNext = null;
    let curr = head;
    
    while (curr) {
        tmpNext = curr.next;
        curr.next = tmpPrev;
        tmpPrev = curr;
        curr = tmpNext;
    }
    
    return tmpPrev;
};